File: 030-Equilibrium Index Finder/EquilibriumIndexFinder.py
def find_equilibrium_index(arr):
    total_sum = 0
    for i in range(len(arr)):
        total_sum += arr[i]

    left_sum = 0
    for i in range(len(arr)):
        total_sum -= arr[i]  # total_sum becomes right sum

        if left_sum == total_sum:
            return i

        left_sum += arr[i]

    return -1


arr = [3, 4, 3, 1, 6]
index = find_equilibrium_index(arr)
print("Equilibrium Index:", index)

File: 030-Equilibrium Index Finder/EquilibriumIndexFinder.cpp
#include<iostream>
#include<vector>
using namespace std;

int findEquilibriumIndex(vector<int>& arr){
    int totalSum = 0;
    for(int i = 0; i < arr.size(); i++){
        totalSum += arr[i];
    }

    int leftSum = 0;
    for(int i = 0; i < arr.size(); i++){
        totalSum -= arr[i];
        if(leftSum == totalSum){
            return i;
        }

        leftSum += arr[i];
    }
    return -1;
}
int main(){
    vector<int> arr = {3, 4, 3, 1, 6};
    int index = findEquilibriumIndex(arr);

    cout<<"Equilibrium Index: " << index << endl; 
    return 0;
}